dashboard/app/assets/stylesheets/application.css.scss

```diff
@@ -560,14 +564,36 @@ html, body {
     background-color: #eee;
     border: 1px solid #ccc;
     display: inline-block;
-    padding: 5px 10px;
+    padding: 5px 5px 5px 10px;
     border-radius: 5px;
   }
-  /* current and last are large dots */
+
+  /* most dots are small */
+  .puzzle_outer_level,
+  .puzzle_outer_assessment {
+    margin: 0 5px 0 -1px;
+    vertical-align: middle;
+    border-radius: 11px;
+    line-height: 1px;
+
+    .level_link {
+      min-width: 1px;
+      width: 1px;
+      height: 1px;
+      border-radius: 9px;
+      padding: 3px;
+      margin: 1px;
+      /* hack to hide the number: */
+      font-size: 0px;
+      color: transparent;
+    }
+  }
+
+  /* current and last dots are large */
   .puzzle_outer_current,
   .puzzle_outer_level.last,
   .puzzle_outer_assessment.last {
-    margin: 0 -1px;
+    margin: 0 5px 0 -1px;
```

Comment:
When I converted to client-side rendering, a bunch of spaces from the HAML versions disappeared.  I fixed them in CSS rather than re-creating the space characters.